### PR TITLE
[Fix](clucene) fix data race and memory leak

### PR DESCRIPTION
**Pull request by zzzxl1993**
1. fix data race and memory leak
2. simple tokenizer lut optimize
3. add clucene index and search test

### CODE REVIEW

File: src/core/CLucene/analysis/Analyzers.cpp [hunk 1]
```diff
@@ -38,23 +38,20 @@ Token *CharTokenizer<char>::next(Token *token) {
                 return NULL;
         } else
             c = ioBuffer[bufferIndex++];
-        if (isTokenChar(c)) {// if it's a token TCHAR
```
Comment (airborne12) [inline src/core/CLucene/analysis/Analyzers.cpp:41]: change isTokenChar to is_alnum will change other tokenizer's behavior

File: src/core/CLucene/analysis/Analyzers.cpp [hunk 2]
```diff
@@ -38,23 +38,20 @@ Token *CharTokenizer<char>::next(Token *token) {
...
+        if (is_alnum(c)) {// if it's a token TCHAR
 
             if (length == 0)// start of token
                 start = offset - 1;
 
-            //buffer[length++] = normalize(c);          // buffer it, normalized
-            buffer[length++] = c;
+            buffer[length++] = to_lower(c);          // buffer it, normalized
```
Comment (airborne12) [inline src/core/CLucene/analysis/Analyzers.cpp:46]: same as isTokenchar's problem

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by adonis0147 — APPROVED** [review]
LGTM

**Pull request merged**